### PR TITLE
Issue #6902: format xdoc/javadoc files to follow rule of LineLength

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6902

Changes only in the config, comments and tests.

### CODE REVIEW

File: config/checkstyle_checks.xml [hunk 1]
```diff
@@ -110,7 +110,7 @@
     <property name="message" value="Line should not be longer than 100 symbols"/>
   </module>
   <module name="RegexpSingleline">
-    <property name="format" value="^(?!(.*href=&quot;|.*http)).{101,}$"/>
+    <property name="format" value="^(?!(.*href=&quot;[^&quot;]+&quot;&gt;|.*http)).{101,}$"/>
```
Comment (rnveach) [inline config/checkstyle_checks.xml]: what about enforcing `<` start the line for the `<a`? If the link is long enough, I think it makes sense it be completely alone on the line.

File: src/xdocs/google_style.xml [hunk 2]
```diff
@@ -21,14 +21,14 @@
...
-                    >cached page</a>),
+                        Google Java Style</a>(
+                    <a href="styleguides/google-java-style-20180523/javaguide.html">
+                    cached page</a>),
                     version of 23 May 2018, current as of 07 May 2019
                 </p>
 
                 <p>
-                    <a href="http://checkstyle.sourceforge.net/reports/google-style/guava/">
+                    <a href="http://checkstyle.org/reports/google-style/guava/">
```
Comment (pbludov) [inline src/xdocs/google_style.xml]: This change is not related to this issue.

File: config/checkstyle_checks.xml (hunk 1 above)
Comment (pbludov) [reply config/checkstyle_checks.xml]: done. Now there is many changes in `google_style.xml`

File: src/xdocs/google_style.xml (hunk 2 above)
Comment (romani) [reply src/xdocs/google_style.xml]: "reports" are only located in sourceforge.io , please use sourceforge.io

File: src/xdocs/releasenotes.xml [hunk 3]
```diff
@@ -5847,7 +5847,8 @@ used in the MT mode.
           move non-compilable input to resources-noncompilable. Author: Vladislav Lisetskii
         </li>
         <li>
-          Minor update in <a href="config_coding.html#FallThrough">FallThroughCheck</a> xdoc and javadoc. Author: Vladislav Lisetskii
```
Comment (rnveach) [inline src/xdocs/releasenotes.xml]: @romani I thought release notes was not subject to this validation or am I wrong?
I know we are still fixing the utility to be better on line lengths.

Comment (romani) [reply src/xdocs/releasenotes.xml]: We can fix links, but it is not good to enforce this in build, nobody review release notes diff, so validation of length is not required. Release notes generation tool should generate content in format we want.

File: src/xdocs/google_style.xml (hunk 2 above)
Comment (pbludov) [reply src/xdocs/google_style.xml]: done

File: src/xdocs/releasenotes.xml (hunk 3 above)
Comment (rnveach) [reply src/xdocs/releasenotes.xml]: @pbludov Can you confirm if this validation is being enforced on this file?

Comment (pbludov) [reply src/xdocs/releasenotes.xml]: done. Added suppression for `releasenotes.xml`, the file itself reverted.

**Comment by rnveach** [comment]
@romani if you are ok, please merge.
